go/chat/teams.go

```diff
@@ -55,3 +56,80 @@ func teamToNameInfo(ctx context.Context, team *teams.Team, vis chat1.TLFVisibili
 	}
 	return res, nil
 }
+
+type ImplicitTeamsNameInfoSource struct {
+	globals.Contextified
+	utils.DebugLabeler
+}
+
+func NewImplicitTeamsNameInfoSource(g *globals.Context) *ImplicitTeamsNameInfoSource {
+	return &ImplicitTeamsNameInfoSource{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "ImplicitTeamsNameInfoSource", false),
+	}
+}
+
+func (t *ImplicitTeamsNameInfoSource) Lookup(ctx context.Context, name string, vis chat1.TLFVisibility) (res types.NameInfo, err error) {
+	// check if name is prefixed
+	if strings.HasPrefix(name, keybase1.ImplicitTeamPrefix) {
+		return t.lookupInternalName(ctx, name, vis)
+	}
+
+	teamID, teamName, err := teams.LookupImplicitTeam(ctx, t.G().ExternalG(), name, vis == chat1.TLFVisibility_PUBLIC)
```

Comment:
I'm a little worried about many team loads we are doing here. I wonder if adding a `LookupByID` might be a good choice?

Reply:
Decided to punt on this for different PR.